[DOCS] Adds brand guide and updates team calls information in contributing (#9390)
ethereum/solidity: docs/brand-guide.rst

programming language.

- In ways that confuse the community as to whether the Solidity
programming language is open-source and free to use.

Solidity Logo License
=====================

.. image:: https://i.creativecommons.org/l/by/4.0/88x31.png
  :width: 88
  :alt: Creative Commons License

The Solidity logo is distributed and licensed under a `Creative Commons
Attribution 4.0 International License <http://creativecommons.org/licenses/by/4.0/>`_.

This is the most permissive Creative Commons license and allows reuse
and modifications for any purpose.

You are free to:

- **Share** — Copy and redistribute the material in any medium or format.

- **Adapt** — Remix, transform, and build upon the material for any
purpose, even commercially.

Under the following terms:

- **Attribution** — You must give appropriate credit, provide a link to
the license, and indicate if changes were made. You may do so in any
reasonable manner, but not in any way that suggests the the Solidity
core team endorses you or your use.

When using the Solidity logo, please respect the Solidity logo guidelines.

Solidity Logo Guidelines
========================

> You can find the official Solidity logo SVG `here <https://github.com/ethereum/solidity/blob/develop/docs/logo.svg>`_.

Reviewer: I think we can use:
```suggestion
.. literalinclude:: logo.svg
   :language: svg
```

####################
Solidity Brand Guide
####################

This brand guide features information on Solidity's brand policy and
logo usage guidelines.

The Solidity Brand
==================

The Solidity programming language is an open-source, community project
governed by a core team. The core team is sponsored by the `Ethereum
Foundation <https://ethereum.foundation/>`_.

This document aims to provide information about how to best use the
Solidity brand name and logo.

We encourage you to read this document carefully before using the
brand name or the logo. Your cooperation is highly appreciated!

Solidity Brand Name
===================

"Solidity" should be used to refer to the Solidity programming language
solely.

> Please don't use "Solidity":

Reviewer 2: ```suggestion
Please do not use "Solidity":
```
I think in general we should avoid contractions in the docs. Maybe we actually have a rule for that in the guideline

- To refer to any other programming language.

- In a way that is misleading or may imply association of unrelated
modules, tools, documentation, or other resources with the Solidity

> programming language.

Reviewer 2: @franzihei for some reason the docs compiler doesn't like when a single item in a bullet list has a line break. So this should be all a single line.

Reviewer: I think all it wants is not having the empty lines between the points.

Reviewer: It also seems to be happy if you indent the new lines with 2 spaces.

Reviewer 2: Ah ok, makes sense

- In ways that confuse the community as to whether the Solidity
programming language is open-source and free to use.

Solidity Logo License
=====================

.. image:: https://i.creativecommons.org/l/by/4.0/88x31.png
  :width: 88
  :alt: Creative Commons License

The Solidity logo is distributed and licensed under a `Creative Commons
Attribution 4.0 International License <http://creativecommons.org/licenses/by/4.0/>`_.

This is the most permissive Creative Commons license and allows reuse
and modifications for any purpose.

You are free to:

- **Share** — Copy and redistribute the material in any medium or format.

- **Adapt** — Remix, transform, and build upon the material for any

> purpose, even commercially.

Reviewer 2: Same here

Under the following terms:

- **Attribution** — You must give appropriate credit, provide a link to

> the license, and indicate if changes were made. You may do so in any

Reviewer 2: And here

reasonable manner, but not in any way that suggests the the Solidity
core team endorses you or your use.

When using the Solidity logo, please respect the Solidity logo guidelines.

Solidity Logo Guidelines
========================

> .. literalinclude:: logo.svg

Reviewer 2: It's also complaining about this line, but I'm not sure what the intention here is

Reviewer: We should just use `.. image:: logo.svg` instead.

Reviewer: Width `width: 128`, 160, or 256 it looks nice.

   :language: svg

Please do not:

> - Change the ratio of the logo (don't stretch it or cut it).

Reviewer 2: ```suggestion
- Change the ratio of the logo (do not stretch it or cut it).
```

- Change the colors of the logo, unless it is absolutely necessary.

Credits
=======

> This document was, in parts, derived from the `Python Software

Reviewer: Derived or inspired by? If it was indeed derived, our license (GPL-3.0) must be compatible with their sites.

Author: ![image](https://user-images.githubusercontent.com/41991517/87684718-de735e00-c782-11ea-906b-b81bc7d11dc8.png)

They use CC0 so that should be fine I guess? And most of the stuff I had from them was taken down in the slimmer version now (without the trademark stuff).

Reviewer: Yes, CC0 is fine. What about the Rust link?

Author: ![image](https://user-images.githubusercontent.com/41991517/87770362-3e1f4700-c81f-11ea-8799-7ebcb927e7bb.png)
"dual-licensed" 🤔 but they also state that they derived their guide from the Python guide.

Author: Actually, we might be able to remove the rust-lang link. The most overlap is this part, which I took from Python and adjusted it for Solidity: 

```
In general, we want the word mark "Python" and the Python logos to be used with minimal restriction to refer to the Python programming language.

We do not want these trademarks to be used:

- to refer to any other programming language
- in a way that is misleading or may imply association of unrelated modules, tools, documentation, or other resources with the Python programming language
- in ways that confuse the community as to whether the Python programming language is open source and free to use
```
